### PR TITLE
Add IsomorphicSearchAgent

### PR DESCRIPTION
**Pull request by NikitaZotov**

### CODE REVIEW

File: problem-solver/cxx/exampleModule/agents/IsomorphicSearchAgent.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    logger->Message(ScLog::Type::Info, "Structures have been found");
+  }
+  else
+  {
+    logger->Message(ScLog::Type::Info, "Structures have not been found");
+  }
+
+  utils::AgentUtils::finishAgentWork((ScMemoryContext *) ms_context.get(), questionNode, answer);
```
Comment (ShunkevichDV) [inline problem-solver/cxx/exampleModule/agents/IsomorphicSearchAgent.cpp]: Не очень нравится конструкция (ScMemoryContext *) ms_context.get(), но возможно это проблема utils::

**Review by ShunkevichDV — APPROVED** [review]

**Review by KovalM — APPROVED** [review]

**Review by KovalM — CHANGES_REQUESTED** [review]
We use angular commit convention: https://github.com/angular/angular/blob/master/CONTRIBUTING.md

Rename your commits by git rebase -i HEAD~2

File: problem-solver/cxx/exampleModule/specifications/agent_of_isomorphic_search/ui_menu_isomorphic_search.scsi [hunk 2]
```diff
@@ -0,0 +1,29 @@
+ui_menu_isomorphic_search <- ui_user_command_class_atom; ui_user_command_class_view_kb; ui_one_argument_command_class;;
```
Comment (NikiforovSergei) [inline problem-solver/cxx/exampleModule/specifications/agent_of_isomorphic_search/ui_menu_isomorphic_search.scsi:1]: Do you add this command to any menu in UI?

Comment (NikitaZotov) [reply problem-solver/cxx/exampleModule/specifications/agent_of_isomorphic_search/ui_menu_isomorphic_search.scsi:1]: I had been add this command in pattern UI.

Comment (NikiforovSergei) [reply problem-solver/cxx/exampleModule/specifications/agent_of_isomorphic_search/ui_menu_isomorphic_search.scsi:1]: What do you mean by "pattern ui"?

**Review by NikiforovSergei — APPROVED** [review]

**Review by KovalM — APPROVED** [review]

**Pull request merged**